core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -653,8 +653,7 @@ private boolean isChanged(
   }
 
   private boolean isChanged(TransactionResult latestResult, TransactionResult result) {
-    return !Objects.equals(latestResult.getId(), result.getId())
-        || latestResult.getVersion() != result.getVersion();
+    return !Objects.equals(latestResult.getId(), result.getId());
```

Comment:
Also, use only the transaction ID for comparison in the serializable validation phase.